pkgs/development/python-modules/angr/default.nix

```diff
@@ -1,40 +1,45 @@
 {
   lib,
   stdenv,
-  ailment,
-  archinfo,
-  buildPythonPackage,
-  cachetools,
-  capstone,
-  cffi,
-  claripy,
-  cle,
-  cppheaderparser,
-  dpkt,
   fetchFromGitHub,
-  gitpython,
-  itanium-demangler,
-  mulpyplexer,
-  nampa,
-  networkx,
-  progressbar2,
-  protobuf,
-  psutil,
-  pycparser,
-  pyformlang,
+  fetchPypi,
+  python,
   pythonOlder,
-  pyvex,
-  rich,
-  rpyc,
-  setuptools,
-  sortedcontainers,
-  sqlalchemy,
-  sympy,
   unicorn,
-  unique-log-filter,
+  unicorn-emu,
+  cmake,
+  pkg-config,
+  cctools,
+  darwin,
+  pkgs,
 }:
 
-buildPythonPackage rec {
+let
+  unicorn-emu-for-angr = unicorn-emu.overrideAttrs (
+    self: super: rec {
+      version = "2.0.1.post1";
+
+      src = fetchFromGitHub {
+        owner = "unicorn-engine";
+        repo = self.pname;
+        rev = version;
+        hash = "sha256-Jz5C35rwnDz0CXcfcvWjkwScGNQO1uijF7JrtZhM7mI=";
+      };
+    }
+  );
+
+  packageOverrides = self: super: {
+    unicorn = (super.unicorn.override { unicorn = unicorn-emu-for-angr; }).overrideAttrs (
+      self: super: { patches = [ ./unicorn-support-python3.12.patch ]; }
+    );
+  };
+
+  overriddenPython = python.override {
+    self = overriddenPython;
+    inherit packageOverrides;
+  };
+in
```

Comment:
`angr` is a package and not an application. We agreed that overriding `unicorn` is the last resort as overriding module inside pythonPackages is a tricky thing because there are side effects.

The override should be a simple as possible (like the one we used in the [past](https://github.com/NixOS/nixpkgs/commit/e11030af6e7ea135291ebe151879bd1259e2ba73)).